chainio/elcontracts/reader.go

```diff
@@ -60,29 +66,42 @@ type ELReader interface {
 }
 
 type ELChainReader struct {
-	logger            logging.Logger
-	elContractsClient clients.ELContractsClient
-	ethClient         eth.EthClient
+	logger                  logging.Logger
+	slasher                 slasher.ContractSlasherCalls
+	delegationManager       delegationmanager.ContractDelegationManagerCalls
+	strategyManager         strategymanager.ContractStrategyManagerCalls
+	blsPubkeyCompendium     blspkcompendium.ContractBLSPublicKeyCompendiumCalls
+	blsPubKeyCompendiumAddr common.Address
+	ethClient               eth.EthClient
 }
 
 // forces EthReader to implement the chainio.Reader interface
 var _ ELReader = (*ELChainReader)(nil)
 
 func NewELChainReader(
```

Comment:
This constructor has become very big. as in what if someone just want to initialize reader and not writer? 
two ways we can handle
1) We make this method private and force users to use constructor
2) We have another method which just accepts few contract addresses and logger, ethclient and just give reader back

I prefer 2 since it's flexible. 


same for writer/subscriber

Reply:
Don’t get your question. This is just the reader constructor. Why are you asking what happens if they want to not build Writer? Was this comment meant to be about constructor.go instead of reader.go?

Reply:
Your 2 sounds like what I was proposing in https://github.com/Layr-Labs/eigensdk-go/pull/11

Reply:
Added EL constructors, for eg https://github.com/Layr-Labs/eigensdk-go/pull/73/commits/3081f8a420ab2cb4608c875b187ff4977918a2e7#diff-78f2b53998897f8fe40be2a3043dec1f342d78f98c9ef5b5abf867a8160fe938R102